specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -12541,38 +12614,24 @@
           "type": "string",
           "title": "The version of the Azure Virtual Machines Marketplace image.",
           "description": "A value of 'latest' can be specified to select the latest version of an image. If omitted, the default is 'latest'."
+        },
+        "virtualMachineImageId": {
+          "type": "string",
+          "title": "The ARM resource identifier of the virtual machine image. Computes nodes of the pool will be created using this custom image. This is of the form /subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/images/{imageName}",
+          "description": "This property is mutually exclusive with other ImageReference properties. The virtual machine image must be in the same region and subscription as the Azure Batch account. For information about the firewall settings for Batch node agent to communicate with Batch service see https://docs.microsoft.com/en-us/azure/batch/batch-api-basics#virtual-network-vnet-and-firewall-configuration."
         }
       },
-      "required": [ "publisher", "offer", "sku" ],
-      "title": "A reference to an Azure Virtual Machines Marketplace image. To get the list of all imageReferences verified by Azure Batch, see the 'List supported node agent SKUs' operation."
+      "title": "A reference to an Azure Virtual Machines Marketplace image or a custom Azure Virtual Machine image. To get the list of all image references verified by Azure Batch, see the 'List node agent SKUs' operation."
```

Comment:
should 'all image references verified by Azure Batch' but modified to refer explicitly to Marketplace images?  (Perhaps redundant since we can never verify a custom image.)  Have we updated the 'List node agent SKUs' documentation to reflect this or is it worded in such a way that it doesn't need any changes?

Reply:
I've improved both this documentation as well as the documentation inside the `NodeAgentSku` type to clarify that the verified image references are always from the Azure Marketplace.